sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConsumer.cs

```diff
@@ -336,6 +418,486 @@ private void CloseLink(ReceivingAmqpLink link)
+            // MessagingEventSource.Log.MessageDeferStop(ClientId);
+        }
+
+
+        /// <summary>Indicates that the receiver wants to defer the processing for the message.</summary>
+        ///
+        /// <param name="message">The lock token of the <see cref="ServiceBusMessage" />.</param>
+        /// <param name="timeout"></param>
+        /// <param name="propertiesToModify">The properties of the message to modify while deferring the message.</param>
+        ///
+        internal virtual Task DeferInternal(
+            ServiceBusReceivedMessage message,
+            TimeSpan timeout,
+            IDictionary<string, object> propertiesToModify = null)
+        {
+            var lockTokens = new[] { new Guid(message.LockToken) };
+            if (lockTokens.Any(lt => _requestResponseLockedMessages.Contains(lt)))
+            {
+                return DisposeMessageRequestResponseAsync(
+                    lockTokens,
+                    timeout,
+                    DispositionStatus.Defered,
+                    _isSessionReceiver,
+                    SessionId,
+                    propertiesToModify);
+            }
+            return DisposeMessagesAsync(lockTokens, GetDeferOutcome(propertiesToModify), timeout);
+        }
+
+        /// <summary>
+        /// Abandons a <see cref="ServiceBusReceivedMessage"/>. This will make the message available again for processing.
+        /// </summary>
+        ///
+        /// <param name="message">The <see cref="ServiceBusReceivedMessage"/> to abandon.</param>
+        /// <param name="propertiesToModify">The properties of the message to modify while abandoning the message.</param>
+        /// <param name="cancellationToken"></param>
+        ///
+        /// <remarks>
+        /// Abandoning a message will increase the delivery count on the message.
+        /// This operation can only be performed on messages that were received by this receiver
+        /// when <see cref="ReceiveMode"/> is set to <see cref="ReceiveMode.PeekLock"/>.
+        /// </remarks>
+        public override async Task AbandonAsync(
+            ServiceBusReceivedMessage message,
+            IDictionary<string, object> propertiesToModify = null,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotClosed(IsClosed, nameof(ServiceBusReceiver));
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            try
+            {
+                await _retryPolicy.RunOperation(
+                    async (timeout) => await AbandonInternalAsync(message, timeout, propertiesToModify).ConfigureAwait(false),
+                    EntityName,
+                    ConnectionScope,
+                    cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                // MessagingEventSource.Log.MessageAbandonException(ClientId, exception);
+                throw exception;
```

Comment:
Should this just be `throw` to preserve the original state and metadata?

Reply:
Leaving just the one comment, but the same pattern exists elsewhere in the class.

Reply:
This stuff will be fixed in a separate PR that will do a sweep of all exceptions.